### PR TITLE
week 3: diogogm-diogotc scientific paper proposal

### PR DESCRIPTION
**Pull request by diogotcorreia**
# Assignment Proposal

## Title

_Microservices: Architecting for Continuous Delivery and DevOps_

## Names and KTH ID

- Diogo Melita (diogogm@kth.se) @d-melita
- Diogo Correia (diogotc@kth.se) @diogotcorreia

## Deadline

Week 3

## Category

Scientific paper

## Description

Paper: [DOI: 10.1109/ICSA.2018.00013](https://doi.org/10.1109/ICSA.2018.00013) - Available through the KTH Library

The paper we intent to present summarizes experiences from Paddy Power about how microservices architecture has been adopted in a Continuous Delivery context. It highlights the substancial advantages of microservices - higher deployability, modifiability, and better resilience - but also discusses the challenges introduced by microservices - including the complexity of managing a larger number of services, evolving service contracts, and handling the diversity of technologies. To address this challenges, practical strategies are presented such as automated CD pipelines and robust contract management. The paper concludes by calling for further research in areas like managing service interactions, refactoring service boundaries, and exploring alternative architectural styles when microservices may not be suitable.

**Relevance**

Continuous deployment is a main topic of DevOps, but it is not always easy to implement for every project. Microservices, as presented in this paper, are an alternative to the common monolithic approach that promises to solve roadblocks in the implementation of CD for software projects.

### CODE REVIEW

**Comment by sofiabobadilla** [comment]
Hi @diogotcorreia, @d-melita

The description should not have a copy-paste abstract of the paper, but instead a summary from you with the main characteristics of it.

The paper seems very relevant, but you must improve the proposal (both on summary and relevance) before I merge the PR.

**Comment by sofiabobadilla** [comment]
Now the proposal is better. 
I'll merge.
PS: Next time ping the TA once you are ready.

Good luck.

**Comment by randomicecube** [comment]
# Feedback from Diogo Gaspar ([dgaspar@kth.se](mailto:dgaspar@kth.se)) and Mateus Marinheiro ([mateusma@kth.se](mailto:mateusma@kth.se))

*We certify that generative AI, incl. ChatGPT, has not been used to write this feedback. Using generative AI without permission is considered academic misconduct.*

Firstly, we want to thank Diogo Correia and Diogo Melita for letting us provide feedback for their presentation – not only did they provide their slides and the paper being discussed, but importantly they also provided us with a script containing the general outline of the contents of the presentation; this allowed us to provide better, more concrete feedback than strictly reading slides and comparing its contents with the paper’s contents, which we greatly appreciate.

As a reference, this feedback starts with listing high-level strengths and weaknesses: all of them are covered later on as well (in Content and Structure/Aesthetics) in greater detail.

## High-Level Strengths

- **Engaging and Approachable Content**: The presentation flows well: it introduces concepts gradually yet steadily, and uses lighthearted elements to ease the audience’s view on the topic.  
- **Great Challenges Section**: The “new challenges and solutions section” was very interestingly put, with the topics being approached from a very deep manner, being of core importance to the usage of microservices.  
- **Highly Relevant Topic**: The aborded topic is a very relevant one especially when it comes to DevOps and CI/CD.  
- **Great Structure/Aesthetics**: The presentation keeps a cohesive structure throughout with slides which aren’t overwhelmed by too much text nor images. The imagery selection also keeps the audience captivated without being distracted by aesthetic differences between slides.

## High-Level Weaknesses

- **Lack of Explanation for Modifiability**: There's a gap in the explanation of the concept of Modifiability in the "Monolithic Applications and Key Characteristics for CD" slide, which is seen as inconsistent with the explanations for other concepts.  
- **Missing Paper Details on Cover**: The cover of the presentation is missing details like the paper’s authors, publishing year, and venue, which (as mentioned by Martin today) need to be added.  
- **Missing code Snipets**: Perhaps nitpicky, but the grading criteria mentions the need for the presentation to cover at least one part that is deeply technical, with readable code snippets; although the “deeply technical” aspect of it seems to be taken care of, the presentation doesn’t contain any code snippets. Although we don’t necessarily see the need for them here, we did find it to be important to mention it, in case it impacts future grading.  
- **Lack of Real-World Example**: Even though it is not mentioned in the grading criteria, we feel like it would be very interesting to mention a real world scenario where the topic would be put into practice (although it may not even be mentioned in the paper itself). For example, Atlassian have a really good [blog post](https://www.atlassian.com/microservices/microservices-architecture/microservices-vs-monolith) where they talk about their experience in changing to a microservices model.

## Content

The presentation flows great, and seems very engaging\! Broadly, the jumps between the contents being discussed seem to be well-grounded, with the concepts being gradually introduced and no major assumptions/previous knowledge having to be held by students. One exception (albeit this may/should be due to some skill issue on our end) is that we didn’t really find a concrete explanation of *Modifiability* upon the introduction of the concept (for example, *Deployability* has one) and we figured it would make sense for its counterpart in the "Monolithic Applications and Key Characteristics for CD" slide to include one as well. Furthermore, we also figured that it could be a good idea to introduce the microservices concept in a slightly different fashion: we get why they’re called *micro*, of course, but why *services*? Are we just dealing with services here? This is very minor (probably even irrelevant), but was a nitpick that we found could be interesting to point out. Maybe this doesn’t even make sense to do (e.g., if the paper itself doesn’t seem to do that), we weren’t quite sure. We also really enjoyed the new challenges and solutions section, with the three core sub-sections (and each’s specific topics) being interesting, far from superficial/too basic – quite the opposite. Finally, we loved the inclusion, in the final slides, of cool, interesting related work: the last one in particular displayed a really interesting visualization of core metric differences between monolithic and microservices architectures, which we found very useful to depict the broader picture for these technologies.

## Structure/Aesthetics

The structure is amazing, we love it\! The sprinkling of lighthearted memes, in contrast with the more “formal” images, allows for the presentation not to “take itself too seriously”, which we found to work really well and be aligned with the faculty’s instructions. It also makes hard concepts seem more approachable which is super helpful if we are dealing with an audience with little background knowledge. We also really enjoyed how in two sections – "Microservices: Observed Benefits", and "Microservices: New Challenges and Solutions" – there’s an initial presentation of all the benefits with a deep dive onto each one of them right after (to be honest, delineating presentation structures isn’t really our *forte*, so this particularity may be a given for most, but we found it to be a really nice point toward the quality of the presentation). Visually, we also loved the image selection: sometimes we may be tempted to be a bit lackluster in this part, but all of them seem to tie into a “major aesthetic”, which contributes to the overall cohesion of the presentation, alongside them giving very good cues on what’s being talked about by themselves. Finally, the take-away message is clear and concise, not leaning towards too vague neither to too specific: in our opinion, it perfectly encapsulates the major benefits of using microservices.

As a late addition to this feedback section, Martin mentioned in today’s lecture that the presentation’s cover should mention the paper’s title/authors/publishing year/venue – the latter three are currently missing (but it’s an easy fix).

## Summary

In summary, we loved the way the paper is approached, and we think the presentation is done in a super interesting, engaging way\! From the introduction of microservices to its benefits and challenges, as well as cool related work and a greatly put take-away message, we think the presentation is lovely, with only a few, small nitpicks and pointers which we thought could be fixed by our colleagues. We hope that the way you deliver your presentation is on the same level as the presentation itself :D. Break a leg\!

**Comment by randomicecube** [comment]
@sofiabobadilla also pinging you because I believe Martin said today that we should probably ping a TA when our comment is done; from here, should we do a PR saying that we submitted our feedback?
Thanks in advance!

**Comment by d-melita** [comment]
Today, @diogotcorreia and I presented the paper in Professor Martin's office. We forgot to mention that we took the feedback from our colleagues, @randomicecube and Mateus, into account before presenting, by adapting our slides and script. We'd like to thank both of them for their feedback, which helped us improve our presentation.
